### DISCUSSION TITLE
How do I make windows open on the same workspace as their parent window?

### DISCUSSION THREAD

**Discussion by omnigenous** [hyprwm/Hyprland#7440] [Q&A] [answered]
Crosspost from https://reddit.com/r/hyprland/comments/1etskvt/how_do_i_make_windows_open_on_the_same_workspace/

---

Very new to hyprland (and wayland too) and I can't figure this out from the wiki, I'd really appreciate any direction.

I am using Unity (the game engine) and it constantly pops up a floating progress bar window when it does stuff. I often switch to a different workspace with a video playing on firefox as it can sometimes take a long time, the problem is this progress bar window often opens and closes again which makes it appear floating over my firefox window instead of staying on the same workspace the main Unity window is on.

Is there a way to make sure all Unity popup windows stay on the same workspace as the main Unity window, without hardcoding it to a specific workspace number? For example, if Unity is on workspace 3, I want all its popups to stay on workspace 3, even if I switch to another workspace.

**Comment by vaxerski** [comment, marked as answer]
not atm exactly, feel free to open an issue

Unless unity is in xwayland, then no. But if it's in native wl, ye
